### ISSUE TITLE
Support boxed types

### ISSUE THREAD

**Issue by aemeryjsa** [realm/realm-java#465]
Currently, only primitive data types are supported, not their equivalent non-primitive types.

It would be nice to be able to distinguish between null and a valid value and write:

```java
public class House extends RealmObject {
    private Integer roomCount;
}
```

Instead, to get the same meaning I need:

```java
public class House extends RealmObject {
    private int roomCount;
    private boolean hasRoomCount;
}
```

The following types could also be supported:
Boolean, Short, Integer, Long, Float, Double

Realm: 0.70.1
Android Studio: 0.8.11
OS: Window 8

**Comment by eliasbagley** [comment]
This would be useful for me. The primitive types cause issues when serializing to JSON since they can't be excluded with nulls

**Comment by dmytrodanylyk** [comment]
+1 from me, I also need this feature

**Comment by damianw** [comment]
Would be very interested in this feature as well.

**Comment by vickychijwani** [comment]
+1. I have exactly the same use-case as @eliasbagley.

**Comment by emanuelez** [comment]
This will be implemented as part of #478
